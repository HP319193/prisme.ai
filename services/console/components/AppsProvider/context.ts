import { createContext, useContext } from 'react';

export interface AppsContext {
  apps: Map<string, Prismeai.App>;
  appInstances: Map<string, Prismeai.DetailedAppInstance[]>;
  getApps: ({
    text,
    page,
    limit,
    workspaceId,
  }: {
    text?: PrismeaiAPI.SearchApps.QueryParameters['text'];
    page?: PrismeaiAPI.SearchApps.QueryParameters['page'];
    limit?: PrismeaiAPI.SearchApps.QueryParameters['limit'];
    workspaceId?: PrismeaiAPI.SearchApps.QueryParameters['workspaceId'];
  }) => Promise<Prismeai.App[] | null>;
  getAppInstances: (
    workspaceId: PrismeaiAPI.ListAppInstances.PathParameters['workspaceId']
  ) => Promise<Prismeai.DetailedAppInstance[]>;
}

export const appsContext = createContext<AppsContext>({
  apps: new Map(),
  appInstances: new Map(),
  getApps: async () => ({} as any),
  getAppInstances: async () => ({} as any),
});

export const useApps = () => useContext(appsContext);

export default appsContext;
