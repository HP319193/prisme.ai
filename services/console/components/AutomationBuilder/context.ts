import { Schema } from '@prisme.ai/design-system';
import { createContext, useContext } from 'react';

export interface AutomationBuilderContext {
  automationId: string;
  addInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  removeInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editCondition: (
    parent: { conditions: Prismeai.Conditions },
    key: string
  ) => void;
  removeCondition: (
    parent: { conditions: Prismeai.Conditions },
    key: string
  ) => void;
  editTrigger: () => void;
  editOutput: () => void;
  getApp: (
    instruction: string
  ) => {
    name: string;
    icon: string;
    instructionName: string;
  };
  instructionsSchemas: [
    string,
    Record<string, Schema & { description?: string }>,
    { icon: string }
  ][];
  getSchema: (name: string) => Schema;
}
export const automationBuilderContext = createContext<AutomationBuilderContext>(
  {
    automationId: '',
    addInstruction() {},
    removeInstruction() {},
    editInstruction() {},
    editCondition() {},
    removeCondition() {},
    editTrigger() {},
    editOutput() {},
    getApp() {
      return {
        name: '',
        icon: '',
        instructionName: '',
      };
    },
    instructionsSchemas: [],
    getSchema() {
      return {} as Schema;
    },
  }
);

export const useAutomationBuilder = () => useContext(automationBuilderContext);

export default automationBuilderContext;
