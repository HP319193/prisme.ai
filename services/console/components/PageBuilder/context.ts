import { createContext } from 'react';
import { Schema } from '@prisme.ai/design-system';
import { BlockInCatalog } from './useBlocks';
import { useContext } from '../../utils/useContext';

export type Block = NonNullable<Prismeai.Page['blocks']>[number];
export type BlocksWithKeys = Map<string, Block>;

export interface PageBuilderContext {
  value: BlocksWithKeys;
  addBlock: (position: number, blockName?: string) => void;
  setEditBlock: (blockId: string) => void;
  removeBlock: (key: string) => void;
  setBlockConfig: (key: string, config: any) => void;
  setBlockSchema: (blockId: string, schema: Schema | null) => void;
  catalog: BlockInCatalog[];
  blocksSchemas: Map<string, Schema | null>;
}

export const context = createContext<PageBuilderContext | undefined>(undefined);

export const usePageBuilder = () => useContext<PageBuilderContext>(context);
