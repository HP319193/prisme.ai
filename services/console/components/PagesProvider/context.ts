import { createContext, Dispatch, SetStateAction, useContext } from 'react';

type WorkspaceId = string;

export interface PagesContext {
  pages: Map<WorkspaceId, Set<Prismeai.Page>>;
  fetchPages: (workspaceId: NonNullable<Prismeai.Workspace['id']>) => void;
  setPages: Dispatch<SetStateAction<Map<string, Set<Prismeai.Page>>>>;
  createPage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  savePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  deletePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    pid: NonNullable<Prismeai.Page['id']>
  ) => Promise<Pick<Prismeai.Page, 'id'>>;
}

export const pagesContext = createContext<PagesContext>({
  pages: new Map(),
  setPages() {},
  fetchPages() {},
  async createPage() {
    return {} as Prismeai.Page;
  },
  async savePage() {
    return {} as Prismeai.Page;
  },
  async deletePage() {
    return {} as Prismeai.Page;
  },
});

export const usePages = () => useContext(pagesContext);

export default usePages;
