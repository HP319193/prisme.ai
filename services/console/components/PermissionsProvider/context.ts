import { createContext, useContext } from 'react';
import UserPermissions = Prismeai.UserPermissions;
import SubjectType = PrismeaiAPI.GetPermissions.Parameters.SubjectType;

export interface PermissionsContext {
  usersPermissions: Map<string, UserPermissions[]>;
  addUserPermissions: (
    subjectType: SubjectType,
    subjectId: string,
    permissions: UserPermissions
  ) => Promise<any>;
  // unshareWorkspaceToUser: (
  //   userEmail: string,
  //   workspaceId: string
  // ) => Promise<any>;
  getUsersPermissions: (
    subjectType: SubjectType,
    subjectId: string
  ) => Promise<UserPermissions[]>;
}

export const workspacesContext = createContext<PermissionsContext>({
  usersPermissions: new Map(),
  // unshareWorkspaceToUser: async () => ({} as any),
  addUserPermissions: async () => ({} as UserPermissions),
  getUsersPermissions: async () => [] as UserPermissions[],
});

export const usePermissions = () => useContext(workspacesContext);

export default workspacesContext;
