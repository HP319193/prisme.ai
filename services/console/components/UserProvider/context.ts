import { createContext, useContext } from 'react';
import { ApiError } from '@prisme.ai/sdk';

export enum OperationSuccess {
  emailSent = 'emailSent',
  passwordReset = 'passwordReset',
}

export interface ApiSuccess {
  type: OperationSuccess;
}
export interface UserContext<T = Prismeai.User | null> {
  user: T;
  loading: boolean;
  error?: ApiError;
  success?: ApiSuccess;
  signin: (email: string, password: string) => Promise<Prismeai.User | null>;
  signup: (
    email: string,
    password: string,
    firstName: string,
    lastName: string
  ) => Promise<Prismeai.User | null>;
  signout: (onServer?: boolean) => void;
  sendPasswordResetMail: (email: string, language: string) => Promise<any>;
  passwordReset: (token: string, password: string) => Promise<any>;
}

export const userContext = createContext<UserContext>({
  user: null,
  loading: false,
  signin: async () => null,
  signup: async () => null,
  signout() {},
  sendPasswordResetMail: async () => null,
  passwordReset: async () => null,
});

export function useUser(throwIfNotExist?: boolean): UserContext;
export function useUser(throwIfNotExist?: true): UserContext<Prismeai.User>;
export function useUser(throwIfNotExist?: boolean) {
  const context = useContext(userContext);

  if (throwIfNotExist && !context.user) {
    throw new Error();
  }

  return context;
}

export default userContext;
