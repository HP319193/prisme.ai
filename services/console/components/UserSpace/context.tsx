import { createContext } from 'react';
import { useContext } from '../../utils/useContext';

export interface UserSpaceConfig {
  /**
   * Main logo alternative
   */
  mainLogo?: {
    /**
     * Logo URL
     */
    url: string;
    /**
     * Attributes to apply on <img /> tag like alt or title
     */
    attrs?: object;
  };
  /**
   * Main URL on main logo link
   * @example: /product/ai-knowledge-chat/assistant?id=6474c0db33959b6283770367
   */
  mainUrl?: string;
  /**
   * The studio can act as a kiosk to keep the user in a URL template
   * @example: /product/ai-knowledge-chat/
   */
  kiosk?: string;
  /**
   * Define if Whats new popover is visible.
   * @default true
   */
  displayWhatsNew?: boolean;
  /**
   * Define if Help popover is visible.
   * @default true
   */
  displayHelp?: boolean;
  /**
   * Define if products popover is visible.
   * @default true
   */
  displayProducts?: boolean;

  /**
   * style object that will be injected inside various sections, allowing to cutomize theme colors ...
   */
  style?: {
    root?: object;
  };

  /**
   * Disable Builder
   */
  disableBuilder?: true;
}

interface UserSpaceProviderContext extends UserSpaceConfig {}
export const userSpaceContext = createContext<
  UserSpaceProviderContext | undefined
>(undefined);
export const useUserSpace = () =>
  useContext<UserSpaceProviderContext>(userSpaceContext);
export default userSpaceContext;
