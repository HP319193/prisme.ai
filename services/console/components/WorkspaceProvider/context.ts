import { createContext, FC, useContext } from 'react';
import { Event, Events, EventsFilters, Workspace } from '@prisme.ai/sdk';

export type Pagination = {
  page: PrismeaiAPI.EventsLongpolling.Parameters.Page;
  limit: PrismeaiAPI.EventsLongpolling.Parameters.Limit;
};
export type EventsByDay = Map<number, Set<Event<Date>>>;
export interface WorkspaceContext {
  workspace: Workspace;
  loading: boolean;
  filters: EventsFilters;
  updateFilters: (newFilters: EventsFilters) => void;
  save: (workspace: Workspace) => void;
  saveSource: (newSource: Workspace) => void;
  events: EventsByDay | 'loading';
  nextEvents: () => void;
  readEvents: Set<string>;
  readEvent: (eventId: string) => void;
  share?: {
    label: string;
    component: FC;
  };
  setShare: (share: WorkspaceContext['share']) => void;
  getAppConfig: (appInstance: string) => any;
  saveAppConfig: (appInstance: string, config: any) => void;
  installApp: (
    workspaceId: PrismeaiAPI.InstallAppInstance.PathParameters['workspaceId'],
    body: PrismeaiAPI.InstallAppInstance.RequestBody
  ) => Promise<Prismeai.AppInstance | null>;
  createAutomation: (
    automation: Prismeai.Automation
  ) => Promise<(Prismeai.Automation & { slug: string }) | null>;
  updateAutomation: (
    slug: string,
    automation: Prismeai.Automation
  ) => Promise<(Prismeai.Automation & { slug: string }) | null>;
  deleteAutomation: (slug: string) => Promise<Prismeai.Automation | null>;
  socket?: Events;
}

export const workspaceProviderContext = createContext<WorkspaceContext>({
  workspace: {} as Workspace,
  loading: false,
  filters: {} as EventsFilters,
  updateFilters: () => {},
  save() {},
  saveSource() {},
  events: 'loading',
  nextEvents() {},
  readEvents: new Set(),
  readEvent() {},
  setShare() {},
  getAppConfig() {},
  saveAppConfig() {},
  installApp: () => ({} as any),
  createAutomation: async () => ({} as Prismeai.Automation & { slug: string }),
  updateAutomation: async () => ({} as Prismeai.Automation & { slug: string }),
  deleteAutomation: async () => ({} as Prismeai.Automation),
});

export const useWorkspace = () => useContext(workspaceProviderContext);

export default workspaceProviderContext;
