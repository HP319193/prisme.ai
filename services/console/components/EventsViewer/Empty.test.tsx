import renderer from 'react-test-renderer';
import EmptyWorkspace from './Empty';
import { workspaceLayoutContext } from '../../layouts/WorkspaceLayout/context';

it('should render', () => {
  const root = renderer.create(
    <workspaceLayoutContext.Provider value={{} as any}>
      <EmptyWorkspace />
    </workspaceLayoutContext.Provider>
  );
  expect(root.toJSON()).toMatchSnapshot();
});
