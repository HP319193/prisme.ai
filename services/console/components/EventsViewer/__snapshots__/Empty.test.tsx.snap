// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="flex flex-1 justify-between flex-col p-28"
>
  <div
    className="flex flex-col flex-1 justify-center w-[17rem]"
  >
    <div
      className="font-bold text-2xl my-2"
    >
      a
    </div>
    <div>
      b
    </div>
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="my-8"
    >
      workspace.empty.subtitle
    </div>
    <div
      className="flex flex-row"
    >
      <button
        className="
        focus:outline-none
        flex flex-col rounded flex-1 m-2 p-4 text-sm text-left justify-between
        transition-all group
        hover:shadow-2xl hover:-translate-y-2
        !ml-0
      "
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F7F8D0",
          }
        }
      >
        <div
          className="flex flex-col"
        >
          <div
            className="font-bold mb-2"
          >
            workspace.empty.suggestions.page.title
          </div>
          <div
            className="relative min-h-[55px]"
          >
            <div
              className="transition-all opacity-100 group-hover:opacity-0"
            >
              workspace.empty.suggestions.page.text
            </div>
            <div
              className=" absolute top-0 flex flex-row mt-4 py-2 px-4 rounded text-3xl bg-[white] transition-all whitespace-nowrap overflow-hidden opacity-0 group-hover:opacity-100 "
              style={
                Object {
                  "color": "#F7F8D0",
                }
              }
            >
              +
            </div>
          </div>
        </div>
      </button>
      <button
        className="
        focus:outline-none
        flex flex-col rounded flex-1 m-2 p-4 text-sm text-left justify-between
        transition-all group
        hover:shadow-2xl hover:-translate-y-2
        undefined
      "
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#FF9261",
          }
        }
      >
        <div
          className="flex flex-col"
        >
          <div
            className="font-bold mb-2"
          >
            workspace.empty.suggestions.automation.title
          </div>
          <div
            className="relative min-h-[55px]"
          >
            <div
              className="transition-all opacity-100 group-hover:opacity-0"
            >
              workspace.empty.suggestions.automation.text
            </div>
            <div
              className=" absolute top-0 flex flex-row mt-4 py-2 px-4 rounded text-3xl bg-[white] transition-all whitespace-nowrap overflow-hidden opacity-0 group-hover:opacity-100 "
              style={
                Object {
                  "color": "#FF9261",
                }
              }
            >
              +
            </div>
          </div>
        </div>
      </button>
      <button
        className="
        focus:outline-none
        flex flex-col rounded flex-1 m-2 p-4 text-sm text-left justify-between
        transition-all group
        hover:shadow-2xl hover:-translate-y-2
        undefined
      "
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E7F6F6",
          }
        }
      >
        <div
          className="flex flex-col"
        >
          <div
            className="font-bold mb-2"
          >
            workspace.empty.suggestions.app.title
          </div>
          <div
            className="relative min-h-[55px]"
          >
            <div
              className="transition-all opacity-100 group-hover:opacity-0"
            >
              workspace.empty.suggestions.app.text
            </div>
            <div
              className=" absolute top-0 flex flex-row mt-4 py-2 px-4 rounded text-3xl bg-[white] transition-all whitespace-nowrap overflow-hidden opacity-0 group-hover:opacity-100 "
              style={
                Object {
                  "color": "#E7F6F6",
                }
              }
            >
              +
            </div>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
